### PR TITLE
vstart.sh: do not support positional arguments of mon,osd,mds

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
@liewegas @alimaredia mind taking a look?

**Comment by liewegas** [comment]
looks fine to me, although i'd still probably lean toward eliminating start_* entirely so that there is just one way to control what is started (MON, MGR, etc. vars). Either way, though!

**Comment by alimaredia** [comment]
@liewegas I agree on having just one way to control what is started. This could make the usage much clearer as well. That's out of the scope of this PR though, perhaps I shoud take it to the mailing list?

**Comment by liewegas** [comment]
i'm only aware of one (old) user (ceph-helpers.sh or something) that i 
removed with teh original vstart mgr changes.  unless someone can point to 
another user i say let's kill it...

**Comment by tchaikov** [comment]
@liewegas could you take another look?

changelog

- rebased against master.
- and removed the start_*

File: src/vstart.sh [hunk 1]
```diff
@@ -57,13 +57,29 @@ export DYLD_LIBRARY_PATH=$CEPH_LIB:$DYLD_LIBRARY_PATH
...
+    CEPH_NUM_MGR=1
+    kill_all=1
+else
+    # and don't start any other daemons if any CEPH_NUM_* is specified.
+    [ -z "$CEPH_NUM_MON" ] && CEPH_NUM_MON=0
+    [ -z "$CEPH_NUM_OSD" ] && CEPH_NUM_OSD=0
+    [ -z "$CEPH_NUM_MDS" ] && CEPH_NUM_MDS=0
+    [ -z "$CEPH_NUM_MGR" ] && CEPH_NUM_MGR=0
+    kill_all=0
+fi
```
Comment (liewegas) [inline src/vstart.sh]: what's the motivation for this?  is this for running without -n?

FWIW I usually run vstart by override one or two of $MON or $MDS (e.g., MON=1 MDS=0) depending on what i'm testing.  not getting the default values if any of them are specified seems weird.

Comment (tchaikov) [inline src/vstart.sh]: i usually just start MON and OSD, without MDS or MGR, this saves the time to ready and launch these daemons, and also i don't compile MDS by default.

Comment (tchaikov) [inline src/vstart.sh]: no, "-k" is the default behavior. the CEPH_NUM_* will be set using the stored number in ceph.conf afterwards if not "-n".

if you feel strong about this, i will instead make them **override** the default values.

**Comment by liewegas** [comment]
oh that's not so bad actually.  I usually do MON=1 OSD=1 MDS=0; instead 
i'd do MON=1 OSD=1 MGR=1 to get the same result.  no real preference, 
then.  @yehudasa?

**Comment by tchaikov** [comment]
@liewegas, please see 	e744327, this behavior is more backward compatible with existing tests.

**Comment by liewegas** [comment]
oh, this is also in wip-mgr-commands too, see 
47cc059d31a112d57690a7b84e042d8eb2b25d94

**Comment by tchaikov** [comment]
okay, i will incorporate 47cc059 and make the numbers override the default settings. actually, much simpler in vstart.sh that way.

**Comment by tchaikov** [comment]
retest this please

**Comment by tchaikov** [comment]
@liewegas fixed and repushed.

**Comment by liewegas** [comment]
lgtm.. ok to merge?

**Comment by tchaikov** [comment]
@liewegas thanks for the review, merging!

**Pull request merged**